### ISSUE TITLE
ClippingNode works incorrectly on some Android devices with Snapdragon 425

### ISSUE THREAD

**Issue by heavos** [cocos2d/cocos2d-x#20482]
- cocos2d-x version: V3.17.2, V4.0
- devices test on: (all devices have Snapdragon 425 and Adreno 308)
   - tested myself: Samsung Galaxy J4+ (since android 9)
   - according to users' feedback: Moto g(6) Play, Samsung Galaxy J6+, Samsung Galaxy Tab A 8.0 2018 and more
- developing environments
   - OS: Windows 10
   - NDK version: 21.0.6113669

Steps to Reproduce (assuming that you use smartphone with Snapdragon 425):
1. Clone [cocos2d-x](https://github.com/cocos2d/cocos2d-x/tree/v3) (in this case it's gonna be V3.17.2 but bug also reproduces on V4.0)
2. Build ${COCOS_PATH}/tests/cpp-tests
3. In the app go to category 28:Node: Clipping
4. In this category tests 2, 5, 7, 8, 12, 13, 14, 15 will fail [like this](https://discuss.cocos2d-x.org/t/clippingnode-doesnt-work-with-alpha-on-some-specific-devices-with-android-9/47355/21)

This issue is based on [this discussion](https://discuss.cocos2d-x.org/t/clippingnode-doesnt-work-with-alpha-on-some-specific-devices-with-android-9/47355) on cocos forum. Tests that fail use sprites with alpha channel as stencil. Somehow alpha channel gets ignored and stencil clips everything behind sprite if **setInverted(false)** and shows everything if **setInverted(true)**.

Firstly my test device (Samsung Galaxy J4+) worked fine with ClippingNode, but after upgrading it to Android 9 this bug started to occur. Based on users feedback and screenshots I also found that similar bug occurs on other devices with the same processor. But not all of them had android 9, there were other versions as well.

I tried to dig into it myself... well... unsuccessfully. But I found something that might be useful.
I profiled 3 devices with [Adreno GPU Profiler](https://developer.qualcomm.com/software/adreno-gpu-profiler). Samsung Galaxy J4+ (Snapdragon 425) which has incorrect behavior and two other Snapdragon devices (Google Pixel 2016 with Snapdragon 821 and Pocophone F1 with Snapdragon 845) on which ClippingNode works fine.

Using Scrubber OpenGL ES in this program I got 3 logs from test 7.
[Samsung Galaxy J4+ Snapdragon 425.csv](https://github.com/cocos2d/cocos2d-x/files/4354852/425.csv.txt)
[Google Pixel 2016 Snapdragon 810.csv](https://github.com/cocos2d/cocos2d-x/files/4354853/810.csv.txt)
[Pocophone F1 Snapdragon 845.csv](https://github.com/cocos2d/cocos2d-x/files/4354854/845.csv.txt)
![AdrenoGPUProfilerS425](https://user-images.githubusercontent.com/62338669/77073453-5b264380-6a00-11ea-8229-3760db478f96.png)

Using Beyond Compare I compared this 3 csv files and found this.
```
Line 95:
S425: glBindVertexArray( array =**49**)
S821: glBindVertexArray( array =**50**)
S845: glBindVertexArray( array =**50**)

Line 98:
S425: glDrawElements( mode =GL_TRIANGLES  count =150  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**86**)
S821: glDrawElements( mode =GL_TRIANGLES  count =150  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**88**)
S845: glDrawElements( mode =GL_TRIANGLES  count =150  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**88**)

Line 115:
S425: glBindBuffer( target =GL_ARRAY_BUFFER  buffer =**85**)
S821: glBindBuffer( target =GL_ARRAY_BUFFER  buffer =**87**)
S845: glBindBuffer( target =GL_ARRAY_BUFFER  buffer =**87**)

Line 119:
S425: glBindBuffer( target =GL_ELEMENT_ARRAY_BUFFER  buffer =**86**)
S821: glBindBuffer( target =GL_ELEMENT_ARRAY_BUFFER  buffer =**88**)
S845: glBindBuffer( target =GL_ELEMENT_ARRAY_BUFFER  buffer =**88**)

Line 120:
S425: glDrawElements( mode =GL_TRIANGLES  count =150  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**86**)
S821: glDrawElements( mode =GL_TRIANGLES  count =150  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**88**)
S845: glDrawElements( mode =GL_TRIANGLES  count =150  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**88**)

Line 122:
S425: glDrawElements( mode =GL_TRIANGLES  count =270  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**88**)
S821: glDrawElements( mode =GL_TRIANGLES  count =270  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**90**)
S845: glDrawElements( mode =GL_TRIANGLES  count =270  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**90**)

Line 128:
S425: glBindBuffer( target =GL_ARRAY_BUFFER  buffer =**87**)
S821: glBindBuffer( target =GL_ARRAY_BUFFER  buffer =**89**)
S845: glBindBuffer( target =GL_ARRAY_BUFFER  buffer =**89**)

Line 132:
S425: glBindBuffer( target =GL_ELEMENT_ARRAY_BUFFER  buffer =**88**)
S821: glBindBuffer( target =GL_ELEMENT_ARRAY_BUFFER  buffer =**90**)
S845: glBindBuffer( target =GL_ELEMENT_ARRAY_BUFFER  buffer =**90**)

Line 133:
S425: glDrawElements( mode =GL_TRIANGLES  count =270  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**88**)
S821: glDrawElements( mode =GL_TRIANGLES  count =270  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**90**)
S845: glDrawElements( mode =GL_TRIANGLES  count =270  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**90**)

Line 160:
S425: glDrawElements( mode =GL_TRIANGLES  count =24  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**81**)
S821: glDrawElements( mode =GL_TRIANGLES  count =24  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**83**)
S845: glDrawElements( mode =GL_TRIANGLES  count =24  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**83**)

Line 172:
S425: glBindBuffer( target =GL_ELEMENT_ARRAY_BUFFER  buffer =**81**)
S821: glBindBuffer( target =GL_ELEMENT_ARRAY_BUFFER  buffer =**83**)
S845: glBindBuffer( target =GL_ELEMENT_ARRAY_BUFFER  buffer =**83**)

Line 173:
S425: glDrawElements( mode =GL_TRIANGLES  count =24  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**81**)
S821: glDrawElements( mode =GL_TRIANGLES  count =24  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**83**)
S845: glDrawElements( mode =GL_TRIANGLES  count =24  type =GL_UNSIGNED_SHORT  indices =0x0  CurrentElementArrayBinding =**83**)
```
Basically, what I tried to find is how "correct" device differs from "incorrect" one. I selected lines where correct devices where the same and incorrect was different. So, this is the results. I don’t know if this information will be useful or not, but here it is.

**Comment by chenfubi** [comment]
unfortunately, I also meet this bugs with cocos2dx-3.16. I find it is discard in the glhl is not working
